src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlAuthenticationClaimResource.cs

```diff
@@ -0,0 +1,140 @@
+        private readonly string _dnsAddress;
+        [DataMember]
+        private readonly string _ipAddress;
+
+        [OnDeserialized]
+        private void OnDeserialized(StreamingContext ctx)
+        {
+            if (string.IsNullOrEmpty(_authenticationMethod))
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(_authenticationMethod));
+            if (AuthorityBindings == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(AuthorityBindings));
+        }
+
+        public SamlAuthenticationClaimResource(
+            DateTime authenticationInstant,
+            string authenticationMethod,
+            string dnsAddress,
+            string ipAddress
+            )
+        {
+            if (string.IsNullOrEmpty(authenticationMethod))
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(authenticationMethod));
+
+            _authenticationInstant = authenticationInstant;
+            _authenticationMethod = authenticationMethod;
+            _dnsAddress = dnsAddress;
+            _ipAddress = ipAddress;
+            AuthorityBindings = (new List<SamlAuthorityBinding>()).AsReadOnly();
+        }
+
+        public SamlAuthenticationClaimResource(
+            DateTime authenticationInstant,
+            string authenticationMethod,
+            string dnsAddress,
+            string ipAddress,
+            IEnumerable<SamlAuthorityBinding> authorityBindings
+            )
+            : this(authenticationInstant, authenticationMethod, dnsAddress, ipAddress)
+        {
+            if (authorityBindings == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException(nameof(authorityBindings)));
+
+            List<SamlAuthorityBinding> tempList = new List<SamlAuthorityBinding>();
+            foreach (SamlAuthorityBinding authorityBinding in authorityBindings)
+            {
+                if (authorityBinding != null)
+                    tempList.Add(authorityBinding);
+            }
+            AuthorityBindings = tempList.AsReadOnly();
+
+        }
+
+        public SamlAuthenticationClaimResource(
+            DateTime authenticationInstant,
+            string authenticationMethod,
+            string dnsAddress,
+            string ipAddress,
+            ReadOnlyCollection<SamlAuthorityBinding> authorityBindings
+            )
+            : this(authenticationInstant, authenticationMethod, dnsAddress, ipAddress)
+        {
+            AuthorityBindings = authorityBindings ?? throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException(nameof(authorityBindings)));
+
+        }
+
+        public DateTime AuthenticationInstant => _authenticationInstant;
+
+        public string AuthenticationMethod => _authenticationMethod;
+
+        public ReadOnlyCollection<SamlAuthorityBinding> AuthorityBindings { get; private set; }
+
```

Comment:
You need to include the private field SamlAuthorityBindings as it has a DataMember attribute which means it will be part of the serialization output.